File: SoftwareTechnologies/LanguageSpecifics/src/Phonebook.java
import java.util.Dictionary;
import java.util.LinkedHashMap;
import java.util.Scanner;

public class Phonebook {
    public static void main(String[] args) {
        Scanner console = new Scanner(System.in);
        LinkedHashMap<String, String> nameAndPhone = new LinkedHashMap<String, String>();

        String command = console.nextLine();
        while (!command.equals("END")){
            String[] split = command.split(" ");

            if (split[0].equals("A")){
                nameAndPhone.put(split[1], split[2]);
            }
            else if (split[0].equals("S")){
                if (nameAndPhone.containsKey(split[1])){
                    System.out.println(split[1] + " -> " + nameAndPhone.get(split[1]));
                }
                else{
                    System.out.println("Contact " + split[1] + " does not exist.");
                }
            }

            command = console.nextLine();
        }
    }
}

File: TechModule/ProgrammingFundamentals/ProgrammingFundamentalsExercises/DictionariesLambdaAndLinqExe/01.Phonebook/Phonebook.cs
using System;
using System.Collections.Generic;
using System.Linq;

class Program
{
    static void Main(string[] args)
    {
        string[] command = Console.ReadLine().Split(' ');
        string action = command[0];
        string name = null;
        string number = null;

        if (action != "END")
        {
            name = command[1];
        }

        if (action == "A")
        {
            number = command[2];
        }

        Dictionary<string, string> dictionary = new Dictionary<string, string>();

        while (action != "END")
        {
            if (action == "A")
            {
                dictionary[name] = number;
            }
            else if (action == "S")
            {
                if (dictionary.ContainsKey(name))
                {
                    Console.WriteLine($"{name} -> {dictionary[name]}");
                }
                else
                {
                    Console.WriteLine($"Contact {name} does not exist.");
                }
            }

            command = Console.ReadLine().Split(' ');
            action = command[0];

            if (action != "END")
            {
                name = command[1];
            }
            if (action == "A")
            {
                number = command[2];
            }
        }
    }
}